app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb

```diff
@@ -6,21 +6,48 @@ def self.event_to_hash(event, ems_id)
       :ems_id     => ems_id,
       :ems_ref    => event[:eventID],
       :timestamp  => event[:time],
+      :message    => event[:message],
       :full_data  => event,
       :username   => event.dig(:user, :email)
     }
 
     _log.debug("event_hash=#{event_hash}")
     case event_hash[:event_type]
+    when /^image/
+      parse_image_event!(event, event_hash)
+    when /^network/
+      parse_network_event!(event, event_hash)
     when /^pvm-instance/
       parse_vm_event!(event, event_hash)
+    when /^volume/
+      parse_volume_event!(event, event_hash)
     end
 
     event_hash
   end
 
+  def self.parse_image_event!(event, event_hash)
+    if ['delete'].include?(event[:action])
```

Comment:
I'm guessing there are going to be more actions to be added here in the future?

Reply:
Or just staying consistent with the other `parse_*_event!` methods?

Reply:
There are quite a few `image` events available today, but unfortunately only `delete` provides enough metadata to trigger a refresh (well, actually I'm parsing the message in the `EventTargetParser` to trigger a refresh when an image import is completed - [ref](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/335/files#diff-d14118ccd17b8c31d6b54be4d044f9d6d9902de875fadefe758a35cae6f62f01R20-R22)). I'm hoping in the future they'll standardize the event responses and we can do more here.